official/transformer/transformer_main.py

```diff
@@ -249,141 +265,185 @@ def train_schedule(
+    if not tf.gfile.Exists(flags_obj.bleu_ref):
+      raise ValueError(
+          "BLEU source file %s does not exist" % flags_obj.bleu_ref)
+    # Make sure that vocab file is in the data directory defined
+    vocab_file_path = os.path.join(flags_obj.data_dir, flags_obj.vocab_file)
+    if not tf.gfile.Exists(vocab_file_path):
+      raise ValueError("Vocabulary file %s does not exist" % vocab_file_path)
+
+
+def main(_):
+  # Load and validate the parsed flags (stored in absl.flags.FLAGS)
+  flags_obj = flags.FLAGS
+  validate_flags(flags_obj)
+
+  # Determine training schedule based on flags.
+  if flags_obj.train_steps is not None:
+    train_eval_iterations = (
+        flags_obj.train_steps // flags_obj.steps_between_evals)
+    single_iteration_train_steps = flags_obj.steps_between_evals
     single_iteration_train_epochs = None
   else:
-    if FLAGS.train_epochs is None:
-      FLAGS.train_epochs = DEFAULT_TRAIN_EPOCHS
-    train_eval_iterations = FLAGS.train_epochs // FLAGS.epochs_between_eval
+    if flags_obj.train_epochs is None:
+      flags_obj.train_epochs = DEFAULT_TRAIN_EPOCHS
+    train_eval_iterations = (
+        flags_obj.train_epochs // flags_obj.epochs_between_evals)
     single_iteration_train_steps = None
-    single_iteration_train_epochs = FLAGS.epochs_between_eval
+    single_iteration_train_epochs = flags_obj.epochs_between_evals
 
-  # Make sure that the BLEU source and ref files if set
-  if FLAGS.bleu_source is not None and FLAGS.bleu_ref is not None:
-    if not tf.gfile.Exists(FLAGS.bleu_source):
-      raise ValueError("BLEU source file %s does not exist" % FLAGS.bleu_source)
-    if not tf.gfile.Exists(FLAGS.bleu_ref):
-      raise ValueError("BLEU source file %s does not exist" % FLAGS.bleu_ref)
+  # Define parameters based on flags
+  if flags_obj.params == "base":
+    params = model_params.TransformerBaseParams
+  elif flags_obj.params == "big":
+    params = model_params.TransformerBigParams
+  # No need to check for other values, because the params flag is an enum.
 
   # Add flag-defined parameters to params object
-  params.data_dir = FLAGS.data_dir
-  params.num_cpu_cores = FLAGS.num_cpu_cores
-  params.epochs_between_eval = FLAGS.epochs_between_eval
+  params.data_dir = flags_obj.data_dir
+  params.num_parallel_calls = flags_obj.num_parallel_calls
+  params.epochs_between_evals = flags_obj.epochs_between_evals
   params.repeat_dataset = single_iteration_train_epochs
-
+  if flags_obj.batch_size is not None:
+    params.batch_size = flags_obj.batch_size
+
+  # Create hooks that log information about the training and metric values
+  train_hooks = hooks_helper.get_train_hooks(
+      flags_obj.hooks,
+      tensors_to_log=TENSORS_TO_LOG,  # used for logging hooks
+      batch_size=params.batch_size  # for ExamplesPerSecondHook
+  )
+  benchmark_logger = logger.config_benchmark_logger(flags_obj.benchmark_log_dir)
+  benchmark_logger.log_run_info("transformer")
```

Comment:
The interface is updated with some extra params, please rebase and add value if needed.